src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -79,11 +79,12 @@
     
     -->
   <ItemGroup>
-    <_AllDirectoriesAbove Include="@(Compile->GetPathsOfAllDirectoriesAbove())" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
+    <_AllDirectoriesAbove Include="@(Compile->GetPathsOfAllDirectoriesAbove())" Condition="'$(DiscoverEditorConfigFiles)' != 'false' or '$(DiscoverGlobalConfigFiles)' != 'false'" />
     <!-- Work around a GetPathsOfAllDirectoriesAbove() bug where it can return multiple equivalent paths when the 
          compilation includes linked files with relative paths - https://github.com/microsoft/msbuild/issues/4392 -->
     <PotentialEditorConfigFiles Include="@(_AllDirectoriesAbove->'%(FullPath)'->Distinct()->Combine('.editorconfig'))" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
-    <EditorConfigFiles Include="@(PotentialEditorConfigFiles->Exists())" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
+    <PotentialEditorConfigFiles Include="@(_AllDirectoriesAbove->'%(FullPath)'->Distinct()->Combine('.globalconfig'))" Condition="'$(DiscoverGlobalConfigFiles)' != 'false'" />
```

Comment:
Should we consider using a different item name, say `GlobalConfigFiles` or `GlobalAnalyzerConfigFiles`, so the same list can be used by NuGet/SDK/tool based providers for explicitly adding global config files?

Reply:
Good idea. I added GlobalEditorConfigFiles as an item group that a user can add to, and added a test to verify.

Reply:
Don't we want to avoid the term `EditorConfig` for global configs?

Reply:
Hmm, good point will switch to `GlobalAnalyzerConfigFiles` don't want just globalconfig as that seems to general.